### PR TITLE
[chore]: add load test workflow

### PR DESCRIPTION
**Pull request by odubajDT**
## Changes

- added load test pipeline from upstream
- added needed Makefile commands
- added oteltestbed for load tests
- added basic load test to confirm the workflow is working
- added benchmarking

### Benchmarks

Benchmark run: https://github.com/Dynatrace/dynatrace-otel-collector/actions/runs/10772992278/job/29871994383?pr=283
Benchmark data: https://github.com/Dynatrace/dynatrace-otel-collector/tree/benchmarks/docs/benchmarks/loadtests

### CODE REVIEW

File: internal/oteltestbedcol/components.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	jaegerreceiver "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/jaegerreceiver"
+	prometheusreceiver "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver"
+	syslogreceiver "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/syslogreceiver"
+	zipkinreceiver "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/zipkinreceiver"
+	pprofextension "github.com/open-telemetry/opentelemetry-collector-contrib/extension/pprofextension"
+	attributesprocessor "github.com/open-telemetry/opentelemetry-collector-contrib/processor/attributesprocessor"
+	resourceprocessor "github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourceprocessor"
+)
+
+func components() (otelcol.Factories, error) {
```
Comment (odubajDT) [inline internal/oteltestbedcol/components.go:31]: the components were adapted to match components which are present in distro collector

File: internal/oteltestbedcol/main.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
+// Code generated by "go.opentelemetry.io/collector/cmd/builder". DO NOT EDIT.
+
+// Program oteltestbedcol is an OpenTelemetry Collector binary.
```
Comment (odubajDT) [inline internal/oteltestbedcol/main.go:3]: file fully backported from upstream

File: internal/oteltestbedcol/main_others.go [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// Code generated by "go.opentelemetry.io/collector/cmd/builder". DO NOT EDIT.
+
+//go:build !windows
+
+package main
```
Comment (odubajDT) [inline internal/oteltestbedcol/main_others.go:5]: file fully backported from upstream

File: internal/oteltestbedcol/main_windows.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
+// Code generated by "go.opentelemetry.io/collector/cmd/builder". DO NOT EDIT.
+
+//go:build windows
+
+package main
```
Comment (odubajDT) [inline internal/oteltestbedcol/main_windows.go:5]: file fully backported from upstream

File: internal/oteltestbedcol/metadata.yaml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+type: oteltestbedcol
+
```
Comment (odubajDT) [inline internal/oteltestbedcol/metadata.yaml:2]: file fully backported from upstream

File: internal/oteltestbedcol/package_test.go [hunk 6]
```diff
@@ -0,0 +1,14 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package main
```
Comment (odubajDT) [inline internal/oteltestbedcol/package_test.go:4]: go.opencensus.io

File: internal/testbed/load/runtests.sh [hunk 7]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+# Copyright The OpenTelemetry Authors
+# SPDX-License-Identifier: Apache-2.0
+
```
Comment (odubajDT) [inline internal/testbed/load/runtests.sh:5]: file fully backported from upstream

File: internal/testcommon/testutil/testutil.go [hunk 8]
```diff
@@ -0,0 +1,133 @@
+package testutil
+
+import (
```
Comment (odubajDT) [inline internal/testcommon/testutil/testutil.go:3]: file fully backported from upstream

File: internal/testcommon/testutil/testutil_test.go [hunk 9]
```diff
@@ -0,0 +1,67 @@
+package testutil
+
+import (
```
Comment (odubajDT) [inline internal/testcommon/testutil/testutil_test.go:3]: file fully backported from upstream

File: internal/oteltestbedcol/components.go [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+
+	factories.Extensions, err = extension.MakeFactoryMap(
+		zpagesextension.NewFactory(),
+		pprofextension.NewFactory(),
+	)
+	if err != nil {
+		return otelcol.Factories{}, err
+	}
+	factories.ExtensionModules = make(map[component.Type]string, len(factories.Extensions))
+	factories.ExtensionModules[zpagesextension.NewFactory().Type()] = "go.opentelemetry.io/collector/extension/zpagesextension v0.108.2-0.20240904075637-48b11ba1c5f8"
```
Comment (mowies) [inline internal/oteltestbedcol/components.go:43]: why are there some commit hash versions in here and some tag versions?

File: .github/workflows/load-tests.yml [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+            ~/.cache/go-build
+          key: loadtest-go-${{ hashFiles('**/go.sum') }}
+
+      - name: Install Dependencies
+        run: make gomoddownload
+  
+      - name: Install Tools
+        run: make install-tools
+
+      - name: Cerate results folder
```
Comment (mowies) [inline .github/workflows/load-tests.yml:86]:
```suggestion
      - name: Create results folder
```

File: .github/workflows/load-tests.yml [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+        run: mkdir -p results && touch results/TESTRESULTS.md
+
+      - name: Download Testbed Binaries
+        uses: actions/download-artifact@v4
+        with:
+          name: testbed-binaries
+          path: bin/
+
+      - name: Make Testbet Binaries executable
```
Comment (mowies) [inline .github/workflows/load-tests.yml:95]:
```suggestion
      - name: Make Testbed Binaries executable
```

File: internal/oteltestbedcol/components.go (hunk 10 above)
Comment (odubajDT) [reply internal/oteltestbedcol/components.go]: mistake, adapted

**Review by mowies — APPROVED** [review]

**Pull request merged**